### PR TITLE
Feature/kien tt/t03 add pagination

### PR DESCRIPTION
**Pull request by truongtrungkien0999**
-Link deloy :https://kientt-job-board.surge.sh/
-Note
New components Page. use Math.ceil to calculate the number of pages.
Add new global variable GET_PAGINATION_JOBS.  To set the number of jobs in 1 page and return new page.
New file unit test Page.  test Page component complete
-Result
image

### CODE REVIEW

**Comment by tyhvb** [comment]
good work @truongtrungkien0999 , could the reformat the code

**Comment by truongtrungkien0999** [comment]
> good work @truongtrungkien0999 , could the reformat the code ? suggest to use the IDE's feature

I reformat the code using the extension Prettier - Code formatter of visual studio code. please check it again

**Comment by tyhvb** [comment]
yeah cool now.
hi @truongtrungkien0999 , i want to tell you advanced part of this, should we apply the api with page & limit param like `mockapi.io/api/job?page=1&limit=10` anytime we navigate to a specific page ? instead of using array slice

**Comment by truongtrungkien0999** [comment]
> yeah cool now. hi @truongtrungkien0999 , i want to tell you advanced part of this, should we apply the api with page & limit param like `mockapi.io/api/job?page=1&limit=10` anytime we navigate to a specific page ? instead of using array slice


i fix the api with page & limit param like mockapi.io/api/job?page=1&limit=10 anytime we navigate to a specific page. please check it again

File: src/api/jobApi.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+import { Job, DEFAULT_LIMIT, API } from "../state/constants/jobConstant";
 
 export const fetchJobsData = async (name: string) => {
-  const { data } = await axios.get<Job[]>(`https://6201d6f6b8735d00174cb605.mockapi.io/api/job`);
-  // const jobs = data.filter((job: Job) => Number(job.id)<=100);
-  // const TOTAL_JOBS = 100;
-  // const jobs = data.filter((job: Job , index :number) => index < TOTAL_JOBS);
-  const jobs = data.slice(0,100);
+  const { data } = await axios.get<Job[]>(API);
+  const jobs = data.slice(0, 100);
```
Comment (tyhvb) [inline src/api/jobApi.ts:6]: Should we slice the jobs here ? i think we should get all of jobs, right ?

File: src/state/actions/jobActions.ts [hunk 2]
```diff
@@ -1,12 +1,24 @@
-import { fetchJobsData } from "../../api/jobApi"
+import { fetchJobsData ,fetchJobsDataByPage} from "../../api/jobApi"
 import { Dispatch } from 'redux'
-import { getJobsAction, getJobsLoading, GET_JOBS, GET_JOBS_ERROR, GET_JOBS_LOADING } from "../constants/jobConstant";
+import { getJobsAction, getJobsLoading, GET_JOBS, GET_JOBS_ERROR, GET_JOBS_LOADING ,GET_PAGINATION_JOBS} from "../constants/jobConstant";
 
 export const getData = (name: string) => async (dispatch: Dispatch<getJobsAction | getJobsLoading>) => {
```
Comment (tyhvb) [inline src/state/actions/jobActions.ts:5]: Do we need to use this function anymore ? if not we can remove it.
also please check and remove unused codes at other places

**Review by tyhvb — COMMENTED** [review]
good work @truongtrungkien0999 , i have left some comments, please check

**Comment by tyhvb** [comment]
At the initial page, we should call the api to get data of page 1, not slice the job data to get that, right ?

File: src/state/actions/jobActions.ts (hunk 2 above)
Comment (truongtrungkien0999) [reply src/state/actions/jobActions.ts]: > At the initial page, we should call the api to get data of page 1, not slice the job data to get that, right ?

I fixed all of them according to your preview. please check it again. thanks

**Pull request merged**